### PR TITLE
ABINIT 7.10.4

### PR DESCRIPTION
**Pull request by mtorrent**
Push Abinit to new version 7.10.4:
- The patch correcting netCDF bugs isn't needed anymore
- Now that a formula is available for ETSF_IO, it can be used for abinit

### CODE REVIEW

File: abinit.rb [hunk 1]
```diff
@@ -60,7 +55,13 @@ def install
...
     end
 
-    if build.with? "netcdf"
+    if build.with? "etsf_io"
+      trio_flavor = "netcdf+etsf_io"
+      args << "--with-etsf-io-incs=-I#{Formula["etsf_io"].opt_include}"
+      args << "--with-etsf-io-libs=-L#{Formula["etsf_io"].opt_lib} -letsf_io_low_level -letsf_io_utils -letsf_io"
+      args << "--with-netcdf-incs=-I#{Formula["netcdf"].opt_include}"
+      args << "--with-netcdf-libs=-L#{Formula["netcdf"].opt_lib} -lnetcdff -lnetcdf"
+    elsif build.with? "netcdf"
```
Comment (iml) [inline abinit.rb]: This won't work if netcdf is disabled by default with `:optional`, as above. Perhaps netcdf should be recommended and etsf_io optional? Or create some option(s) for finer control.

File: abinit.rb [hunk 2]
```diff
@@ -60,7 +55,13 @@ def install
       args << "--with-linalg-libs=-L#{Formula["veclibfort"].opt_lib} -lveclibfort"
     end
 
-    if build.with? "netcdf"
+    if build.with? "etsf_io"
+      trio_flavor = "netcdf+etsf_io"
+      args << "--with-etsf-io-incs=-I#{Formula["etsf_io"].opt_include}"
+      args << "--with-etsf-io-libs=-L#{Formula["etsf_io"].opt_lib} -letsf_io_low_level -letsf_io_utils -letsf_io"
+      args << "--with-netcdf-incs=-I#{Formula["netcdf"].opt_include}"
+      args << "--with-netcdf-libs=-L#{Formula["netcdf"].opt_lib} -lnetcdff -lnetcdf"
```
Comment (iml) [inline abinit.rb]: This won't work if netcdf is disabled by default with `:optional`, as above. Perhaps netcdf should be recommended and etsf_io optional? Or create some option(s) for finer control.

**Comment by mtorrent** [comment]
netcdf is a mandatory dependence of etsf_io; I think this should work.
3 possibilities:
- with-etsf-io : netcdf is already installed
- without-etsf-io, with-netcdf : install netcdf if necessary
- without-etsf-io, without-netcdf: nothing done

**Comment by iml** [comment]
I would make `etsf_io` and `netcdf` both `:recommended` and put the args for each under ifs. If you want to robustify the etsf_io conditional you can add something like 
```
if build.with? "etsf_io"
  fail "Building with etsf_io support requires netcdf" if build.without? "netcdf"

**Comment by mtorrent** [comment]
OK, done!

**Comment by iml** [comment]
Pulled 30947713013. Thanks!

**Pull request closed without merging**